oxidation/libparsec_python/src/crypto.rs

```diff
@@ -52,3 +55,60 @@ impl HashDigest {
         }
     }
 }
+
+#[pyclass]
+#[derive(PartialEq, Eq)]
+pub(crate) struct SecretKey(parsec_api_crypto::SecretKey);
+
+#[pymethods]
+impl SecretKey {
+    #[new]
+    fn new(data: &[u8]) -> PyResult<Self> {
+        match parsec_api_crypto::SecretKey::try_from(data) {
+            Ok(h) => Ok(Self(h)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    #[classmethod]
+    fn generate(_cls: &PyType) -> PyResult<SecretKey> {
+        Ok(SecretKey(parsec_api_crypto::SecretKey::generate()))
+    }
+
+    #[getter]
+    fn secret<'p>(&self, py: Python<'p>) -> PyResult<&'p PyBytes> {
+        Ok(PyBytes::new(py, self.0.as_ref()))
+    }
+
+    pub fn encrypt<'p>(&self, py: Python<'p>, data: &[u8]) -> Result<&'p PyBytes, PyErr> {
```

Comment:
Why a `Result` and not a `PyResult` like other methods?

Reply:
i just forgot to put PyResult instead of Result